src/cursor.rs

```diff
@@ -0,0 +1,171 @@
+use core::fmt;
+use std::{error::Error, sync::Arc};
+
+use crate::{icon::PIXEL_SIZE, platform_impl::PlatformCustomCursor};
+
+#[derive(Debug, Clone)]
+pub struct CustomCursor {
+    pub(crate) inner: Arc<PlatformCustomCursor>,
+}
+
+impl PartialEq for CustomCursor {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.inner, &other.inner)
+    }
+}
```

Comment:
Why did we need this anyway? Doesn't the underlying type support `Partial(Eq)` already?

Reply:
The default Eq would compare the rgba buffers or whatever was inside. I think I needed this pointer comparison at some point but don't anymore. So I dunno if we want this anymore.

Reply:
We should be able to do it via `derive` if the underlying type supports it now. Imo public types should always implement `Partial(Eq)` if possible.

Reply:
Added normal PartialEq and Eq